src/lightning/fabric/plugins/io/xla.py

```diff
@@ -61,4 +62,5 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_optio
             checkpoint = apply_to_collection(checkpoint, (DictConfig, ListConfig), OmegaConf.to_container)
         import torch_xla.core.xla_model as xm
 
-        xm.save({k: v for k, v in checkpoint.items() if k != "callbacks"}, path)
```

Comment:
I'm not 100% sure if the `callback` key skip is still necessary. Let's see what CI thinks about that